src/kernels/common/baseJupyterSession.ts

```diff
@@ -175,44 +174,33 @@ export abstract class BaseJupyterSession implements IJupyterSession {
             return;
         }
 
-        // Start the restart session now in case it wasn't started
-        if (!this.restartSessionPromise) {
-            this.startRestartSession(false);
```

Comment:
The code is better now, previously we'd call `startRestartSesion` and expect `restartSessionPromise` to be initialied and we'd await on that.
Now we just call the method `startRestartSession` and await on the result.